gossip/storage_test.go

```diff
@@ -28,17 +29,34 @@ import (
 )
 
 type testStorage struct {
+	mu          sync.Mutex
```

Comment:
nit: why not just embed the mutex?  
It just makes the locking/unlocking code look nicer.